src/fetchers.ts

```diff
@@ -37,25 +83,36 @@ export async function getPodFetchersByAddressOrEns(identifier: string): Promise<
     address = await multicall.resolveName(Name.name);
   }
 
-  const podId = await Name.getText('podId');
-  if (!podId) throw new Error('No podId on ENS found');
+  let podId = await Name.getText('podId');
+  let controllerAddress;
 
-  // Member token tracks Controller for a given pod ID
-  const memberTokenDeployment = getDeployment('MemberToken', network);
-  const MemberToken = new ethers.Contract(
-    memberTokenDeployment.address,
-    memberTokenDeployment.abi,
-    multicall,
-  );
+  if (podId) {
+    controllerAddress = await getControllerAddressByPodId(podId);
+  } else {
+    // fallback checker for pods without .pod ENS names
+    controllerAddress = await getControllerAddressByPodAddress(address);
+  }
+
+  let controllerDeployment;
+
+  try {
+    // we can not guarantee that the controller adddress is a valid controller
+    controllerDeployment = getControllerByAddress(controllerAddress, network);
+  } catch (err) {
+    throw new Error('Address is not a pod');
```

Comment:
🎨 because we are just grabbing the pod safe's guard, we can't actually confirm its one of our controllers until we call this function